pippo-core/src/main/java/ro/pippo/core/AbstractTemplateEngine.java

```diff
@@ -0,0 +1,173 @@
+ *
+ * To use the convenience methods in this class, implementations must override {@link #init(Application)}
+ * and call this class' implementation before performing any of their initialization.
+ *
+ * @param <CT> configuration target for custom initialization, implementations could use this
+ *
+ * @see TemplateEngine
+ *
+ * @author Ranganath Kini
+ */
+public abstract class AbstractTemplateEngine<CT> implements TemplateEngine {
+
+    private Languages languages;
+    private Messages messages;
+    private Router router;
+    private PippoSettings pippoSettings;
+
+    private String fileExtension;
+    private String templatePathPrefix;
+
+    /**
+     * Performs common initialization for template engines
+     *
+     * Implementations must override this method to do their own template engine specific initialization. To
+     * use the convenience of this class, implementations must invoke this class's implementation before
+     * performing their own initialization.
+     *
+     * @param application reference to the Pippo {@link Application} that can be used to retrieve settings
+     *                     and other settings for the initialization
+     */
+    @Override
+    public void init(Application application) {
+        languages = application.getLanguages();
+        messages = application.getMessages();
+        router = application.getRouter();
+        pippoSettings = application.getPippoSettings();
+
+        fileExtension = pippoSettings.getString(PippoConstants.SETTING_TEMPLATE_EXTENSION, getDefaultFileExtension());
+        templatePathPrefix = pippoSettings.getString(PippoConstants.SETTING_TEMPLATE_PATH_PREFIX, TemplateEngine.DEFAULT_PATH_PREFIX);
+    }
+
+    /**
+     * Performs additional initialization on the specified template engine specific configuration target
+     *
+     * This method serves as a hook for implementation to support additional fine-grained extension to initialize
+     * a template engine specific configuration object.
+     *
+     * Implementations can support this by invoking this method from {@link #init(Application)}.
+     * Supporting this is completely optional.
+     *
+     * The default implementation throws a {@link NotImplementedException}.
+     *
+     * @param application reference to the Pippo {@link Application} the can be used to access configuration details
+     * @param configurationTarget reference to the template engine specific configuration object that needs initialization
+     *
+     * @throws NotImplementedException
+     *          if no additional initialization is supported by the implementation
+     */
+    protected void init(Application application, CT configurationTarget) {
+        throw new NotImplementedException();
+    }
+
+    /**
+     * @see TemplateEngine#renderString(String, Map, Writer)
+     */
+    @Override
+    public abstract void renderString(String templateContent, Map<String, Object> model, Writer writer);
```

Comment:
No need to override this method here. The method is present already in `TemplateEngine`.

Reply:
fixed